packages/core/datadog-sourcemaps.gradle

```diff
@@ -0,0 +1,139 @@
+        def targetName = variant.name.capitalize()
+        def targetPath = variant.dirName
+
+        def bundleTask = tasks.findByName("bundle${targetName}JsAndAssets")
+
+        if (bundleTask == null) {
+            logger.info("Cannot find JS bundle task for target=${targetName}.")
+            return
+        }
+        if (!bundleTask.enabled) {
+            logger.info("JS bundle task for target=${targetName} is not enabled.")
+            return
+        }
+
+        def bundleAssetName = reactConfig.bundleAssetName ?: "index.android.bundle"
+
+        def jsBundleDir = file("$buildDir/generated/assets/react/${targetPath}")
+        def jsBundleFile = file("$jsBundleDir/$bundleAssetName")
+
+        def jsSourceMapsDir = file("$buildDir/generated/sourcemaps/react/${targetPath}")
+        def jsOutputSourceMapFile = file("$jsSourceMapsDir/${bundleAssetName}.map")
+
+        def uploadTask = tasks.create("upload${targetName}Sourcemaps") {
+            group = "datadog"
+            description = "Uploads sourcemaps to Datadog."
+
+            def execCommand = [
+                    "yarn",
+                    "datadog-ci",
+                    "react-native",
+                    "upload",
+                    "--platform",
+                    "android",
+                    "--service",
+                    serviceName,
+                    "--bundle",
+                    jsBundleFile.absolutePath,
+                    "--sourcemap",
+                    jsOutputSourceMapFile.absolutePath,
+                    "--release-version",
+                    releaseVersion,
+                    "--build-version",
+                    buildVersion
+            ]
+
+            doFirst {
+                if (!jsBundleFile.exists()) {
+                    throw new GradleException("JS bundle file doesn't exist, aborting upload.")
+                }
+
+                if (!jsOutputSourceMapFile.exists()) {
+                    throw new GradleException("JS sourcemap file doesn't exist, aborting upload.")
+                }
+
+                def env = [:]
+                if (System.getenv("DATADOG_API_KEY") == null) {
+                    def keyFile = new File(reactRoot, "datadog.credentials")
+                    if (!keyFile.exists() || keyFile.text.isEmpty()) {
+                        throw new GradleException("Cannot find Datadog API key, it is neither" +
+                                " in env, nor in datadog.credentials file.")
+                    }
+                    env["DATADOG_API_KEY"] = keyFile.text
+                }
+
+                logger.info("Going to run sourcemaps upload using the following command:" +
+                        " ${execCommand.join(" ")} and env=${env}")
+                runShellCommand(execCommand, env)
+
+            }
+        }
+
+        uploadTask.dependsOn bundleTask
+        bundleTask.finalizedBy uploadTask
```

Comment:
why this also ? isn't the first dependency enough ?

Reply:
No, not enough. Without `uploadTask.dependsOn bundleTask` there will be an error if user runs upload task, but there is no bundle task run before by some reason. Without `bundleTask.finalizedBy uploadTask` there won't be automatic upload after bundle task is finished.

Reply:
I see, tks for the explanation. That makes sense.